nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -151,17 +172,34 @@ public ValidationResult validate(final String subject, final String value, final
         return propertyDescriptors;
     }
 
-    private ObjectMapper mapper = new ObjectMapper();
+    private ObjectMapper mapper;
 
     //Turn a list of Mongo result documents into a String representation of a JSON array
-    private String buildBatch(List<Document> documents) throws IOException {
-        List<Map> docs = new ArrayList<>();
-        for (Document document : documents) {
-            String asJson = document.toJson();
-            docs.add(mapper.readValue(asJson, Map.class));
+    private String buildBatch(List<Document> documents, String jsonTypeSetting) throws IOException {
+        StringBuilder builder = new StringBuilder();
+        for (int index = 0; index < documents.size(); index++) {
+            Document document = documents.get(index);
+            String asJson;
+            if (jsonTypeSetting.equals(JSON_TYPE_NORMAL)) {
+                asJson = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(document);
+            } else {
+                asJson = document.toJson(new JsonWriterSettings(true));
+            }
+            builder
+                .append(asJson)
+                .append( (documents.size() > 1 && index + 1 < documents.size()) ? ", " : "" );
         }
 
-        return mapper.writeValueAsString(docs);
+        return "[" + builder.toString() + "]"; //mapper.writerWithDefaultPrettyPrinter().writeValueAsString(docs);
```

Comment:
This looks like a dead code comment, if it doesn't inform the line please remove :)